### PR TITLE
Unify react versions in the packages

### PR DESCRIPTION
**Pull request by ross-pfahler**
2 things:
1. a couple instances where these had higher deps verses other packages
2. when running `tsc` with `skipLibCheck: false` there would be multiple conflicting copies of `@types/react` floating around. This attempts to unify on a single version and reduces the number of errors (when running the aformentioned command) from around ~200-300 to ~50.

This fixes a number of linting errors that occur with the `Tab` and `TabList` work. I know this is going to be reconsidered, so I provided workarounds for now.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

React Spectrum

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "@testing-library/react": "^10.0.3",
     "@testing-library/react-hooks": "^3.2.1",
     "@testing-library/user-event": "^10.3.1",
-    "@types/react": "^16.8.0",
+    "@types/react": "16.9.23",
```
Comment (ross-pfahler) [inline package.json:72]: we *may* even be able to drop this. what does everyone think?

Comment (LFDanLu) [reply package.json:72]: Don't we need this for types from React like `HTMLAttributes`, at least that was my understanding from https://www.typescriptlang.org/docs/handbook/react-&-webpack.html#install-our-dependencies? (those instructions are for webpack but I figure the package.json stuff applies universally)

Comment (ross-pfahler) [reply package.json:72]: yes, seems so 👍

**Comment by ross-pfahler** [comment]
This upgrade breaks the linter (or maybe enforces more strongly the linter?)

File: package.json (hunk 1 above)
Comment (ross-pfahler) [inline package.json:72]: this could also be `"16.9.23"` if we want -- let me know.

**Pull request merged**